tests/test_symbolic_primitives.py

```diff
@@ -209,39 +212,51 @@ def test_place_ontop(self, env, prim_gen, apple, pan):
             env.step(action)
         assert apple.states[object_states.OnTop].get_value(pan)
 
-    @pytest.mark.skip(reason="primitives are broken")
-    def test_toggle_on(self, env, prim_gen, stove):
+    def test_toggle_on(self, env, prim_gen, stove, sink):
         assert not stove.states[object_states.ToggledOn].get_value()
         for action in prim_gen.apply_ref(SymbolicSemanticActionPrimitiveSet.TOGGLE_ON, stove):
             env.step(action)
         assert stove.states[object_states.ToggledOn].get_value()
 
-    @pytest.mark.skip(reason="primitives are broken")
-    def test_soak_under(self, env, prim_gen, robot, sponge, sink):
-        water_system = env.scene.get_system("water")
-        assert not sponge.states[object_states.Saturated].get_value(water_system)
-        assert not sink.states[object_states.ToggledOn].get_value()
+        # toggle off the stove after the test is done
+        for action in prim_gen.apply_ref(SymbolicSemanticActionPrimitiveSet.TOGGLE_OFF, stove):
+            env.step(action)
+        assert not stove.states[object_states.ToggledOn].get_value()
 
-        # First toggle on the sink
+        assert not sink.states[object_states.ToggledOn].get_value()
         for action in prim_gen.apply_ref(SymbolicSemanticActionPrimitiveSet.TOGGLE_ON, sink):
             env.step(action)
         assert sink.states[object_states.ToggledOn].get_value()
 
-        # Then grasp the sponge
+        # toggle off the stove after the test is done
+        for action in prim_gen.apply_ref(SymbolicSemanticActionPrimitiveSet.TOGGLE_OFF, sink):
+            env.step(action)
+        assert not sink.states[object_states.ToggledOn].get_value()
+
+    def test_soak_under(self, env, prim_gen, robot, sponge, sink):
+        water_system = env.scene.get_system("water")
+        assert not sponge.states[object_states.Saturated].get_value(water_system)
+        assert not sink.states[object_states.ToggledOn].get_value()
+
+        # First grasp the sponge
         for action in prim_gen.apply_ref(SymbolicSemanticActionPrimitiveSet.GRASP, sponge):
             env.step(action)
         assert robot.states[object_states.IsGrasping].get_value(sponge)
 
+        # Then toggle on the sink
+        sink.states[object_states.ToggledOn].set_value(True)
+        assert sink.states[object_states.ToggledOn].get_value()
+
         # Then soak the sponge under the water
         for action in prim_gen.apply_ref(SymbolicSemanticActionPrimitiveSet.SOAK_UNDER, sink):
             env.step(action)
         assert sponge.states[object_states.Saturated].get_value(water_system)
 
-    # def test_soak_inside():
-    #    pass
+        # toggle off the sink after the test is done
+        sink.states[object_states.ToggledOn].set_value(False)
+        assert not sink.states[object_states.ToggledOn].get_value()
```

Comment:
Same issue as above

Reply:
Recommendation: remove this

Reply:
I was informed that this is just for a speed improvement, and the reset takes care of it, so I'm OK leaving it in.